org.eclipse.scanning.api/src/org/eclipse/scanning/api/points/AbstractGenerator.java

```diff
@@ -32,7 +32,7 @@
 	protected volatile T model; // Because of the validateModel() method
 	
 	protected List<IPointContainer> containers;
-	protected Collection<Object> regions;
+	protected Collection<Object> regions = new ArrayList<Object>();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)